contracts/modules/cover/Cover.sol

```diff
@@ -672,6 +690,14 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return _productTypes.length;
   }
 
+  function isProductDeprecated(uint productId) internal view returns (bool) {
```

Comment:
here i think i wasn't super clear in my original comment i meant to say if we take this check out of isPoolAllowed then it makes sense to make it an internal function

as it's still inside of there and used in 1 place i'd just inline

_products[productId].isDeprecated

because it's only used in 1 place 

My bad if i wasn't clear enough.